### ISSUE TITLE
TextBox Component: Usage of node in label prop derived from TextBox Component within Select/MultiSelect Component

### ISSUE THREAD

**Issue by Manisureal** [Sage/carbon#3569]
### Current behaviour
When using the Label prop(derived from TextBox Component) in Select Component, I can pass an object but this results in a few console errors:
- Failed prop type: Invalid prop `label` of type `object` supplied to `ForwardRef`, expected `string`
- Failed prop type: Invalid prop `label` of type `object` supplied to `SelectTextbox`, expected `string`
- Failed prop type: Invalid prop `label` of type `object` supplied to `Textbox`, expected `string`

### Desired behaviour
Above console errors should not appear when using a node as opposed to a string as per the label prop type validation

### Suggested solution
Textbox Component propType should be extended to allow the usage of a string or a node. In my example use-case, I need the Select Component to have a Clear button inline with the label. The only way to achieve this is to have the label prop outputting a label and also a button.

The same should also be possible for when using the MultiSelect Component.

### Describe alternatives you've considered
Not using the label prop provided by the component and create my own label, however, from an accessibility stand this did not seem right.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 54.3.4
| react-scripts    | 
| React            | 16.13.1
| Browser          | Chrome
| npm              | 6.9.0
| Operating System | MacOS Mojave

**Comment by jamime** [comment]
In this case the `label` must be a string because it is placed within a `<label for="id">` element. In this case we don't want to add it as a child of the label as it will be announced by a screenreader as the label for the input. e.g. `Firstname clear, textbox`.

We will identify an appropriate way to resolve this and provide an example in storybook.

I've asked @tempertemper for some advice.

**Comment by sjohnson-sage** [comment]
Following guidance from @tempertemper the problem that this enhancement would have solved has been avoided therefore we no longer need this functionality added into Carbon. This issue can be closed

**Comment by jamime** [comment]
For future reference, they chose to add an option to the select dropdown that had no value, instead of introducing a clear button.

```jsx
 <option selected>None</option>
```
